### PR TITLE
Bump spring to version 5.3.33

### LINKED ISSUE #13406: CVE-2024-22243 on Spring Framework
### Issue description

Hi, folks. We've been running some applications on Grails 5.3.5 and recently we've been alerted of `CVE-2024-22243`. I noticed that the Spring Web uses a lot the vulnerable component (`UriComponentsBuilder`).

So, I'd like to suggest upgrading the current Spring Version of three dependencies from `5.3.25` to `5.3.32`:

- `org.springframework:spring-web`
- `org.springframework:spring-core`
- `org.springframework:spring-expression`

We overwritten those deps and the application is doing fine, since the upgrade is just some patches.
What are your thoughts about it? Maybe this could be the 5.3.7 release.

### PR DESCRIPTION
**Pull request by guillermocalvo**
Resolves #13406

### CODE REVIEW

**Comment by matrei** [comment]
@guillermocalvo There is now also a [5.3.33 version](https://github.com/spring-projects/spring-framework/releases/tag/v5.3.33). Should we upgrade directly to that instead?

Also, I think you need to add `spring.version=5.3.33` as well as the `springVersion=5.3.33` for the correct version to be resolved in apps. We have done that in `6.1.x` #13205

File: gradle.properties [hunk 1]
```diff
@@ -41,7 +41,7 @@ snakeyamlVersion=2.2
 spockVersion=2.1-groovy-3.0
 springBootVersion=2.7.10
 springLoadedVersion=1.2.8.RELEASE
-springVersion=5.3.26
+springVersion=5.3.32
```
Comment (guillermocalvo) [inline gradle.properties]:
```suggestion
springVersion=5.3.33
spring.version=5.3.33
```

**Comment by matrei** [comment]
@puneetbehl Ready for 5.4.0 release 👍

**Comment by matrei** [comment]
Should probably renamed to "Bump spring to 5.3.33"

**Comment by puneetbehl** [comment]
Although, I have merged it but I think keeping only `springVersion` should be enough. Also, bots like dependabot wouldn't able to bump the versions. I think we should revert that specific change.

**Comment by matrei** [comment]
> Although, I have merged it but I think keeping only `springVersion` should be enough. Also, bots like dependabot wouldn't able to bump the versions. I think we should revert that specific change.

I think, if `spring.version` is not set, the version from the `micronaut-bom` is selected.

**Comment by imranmir** [comment]
@guillermocalvo @puneetbehl can you guys please share the Grails 5.x release version for this fix?

**Comment by puneetbehl** [comment]
This will be shipped with Grails 5.4.0 this week.

**Comment by imranmir** [comment]
Thank you for the response.

**Pull request merged**